system-parachains/coretime/coretime-polkadot/src/migrations.rs

```diff
@@ -0,0 +1,330 @@
+		for (i, (para_id, until)) in POTENTIAL_RENEWALS.iter().enumerate() {
+			// Add the system parachains as an offset - these should come before
+			// the leases.
+			let core_id = i as u16 + 4;
+
+			// This is the entry found in Workplan and PotentialRenewals.
+			let workload = Schedule::truncate_from(Vec::from([ScheduleItem {
+				mask: CoreMask::complete(),
+				assignment: Task(*para_id),
+			}]));
+
+			// Make sure they're not in the leases.
+			assert!(!leases.contains(&LeaseRecordItem { until: *until, task: *para_id }));
+
+			// Ensure they can renew in the next region.
+			assert_eq!(
+				PotentialRenewals::<Runtime>::get(PotentialRenewalId {
+					core: core_id,
+					when: sale_info.region_end + 5040,
+				}),
+				Some(PotentialRenewalRecord {
+					price: 1_000_000_000_000,
+					completion: pallet_broker::CompletionStatus::Complete(workload.clone())
+				})
+			);
+
+			// They should all be in the workplan for next sale.
+			assert_eq!(Workplan::<Runtime>::get((workplan_start, core_id)), Some(workload));
+		}
+
+		// Walk the workplan at timeslice 287565 and make sure there is an entry for every 62 cores.
+		for i in 0..61 {
+			// TODO - finalise
+			assert!(Workplan::<Runtime>::get((287565, i)).is_some());
+		}
+
+		// Ensure we have requested the correct number of cores.
+		assert!(frame_system::Pallet::<Runtime>::read_events_no_consensus().any(|e| {
+			match e.event {
+				crate::RuntimeEvent::Broker(
+					pallet_broker::Event::<Runtime>::CoreCountRequested { core_count },
+				) => {
+					log::info!("{core_count:?}");
+
+					core_count == 62
+				},
+				_ => false,
+			}
+		}));
+
+		Ok(())
+	}
+}
+
+// Incorrect potential renewals in state
+const INCORRECT_RENEWAL_IDS: [(u16, u32); 6] =
+	[(6, 292605), (5, 292605), (13, 292605), (15, 292605), (47, 292605), (44, 292605)];
+
+// Hardcoded para ids and their end timeslice.
+// Taken from https://github.com/SBalaguer/coretime-migration/blob/master/polkadot-output-200924.json
+const POTENTIAL_RENEWALS: [(u32, u32); 5] =
+	[(2048, 283680), (3375, 283680), (3358, 283680), (2053, 283680), (2056, 283680)];
+
+const LEASES: [(u32, u32); 46] = [
+	(2094, 298800),
+	(2040, 298800),
+	(3333, 298800),
+	(2106, 298800),
+	(2093, 298800),
+	(2101, 298800),
+	(3369, 313920),
+	(2000, 313920),
+	(3338, 329040),
+	(2004, 329040),
+	(3344, 344160),
+	(3345, 344160),
+	(3340, 344160),
+	(2031, 344160),
+	(3346, 344160),
+	(2026, 344160),
+	(2006, 344160),
+	(2035, 359280),
+	(2032, 359280),
+	(2025, 359280),
+	(2002, 359280),
+	(2034, 359280),
+	(2012, 359280),
+	(3354, 359280),
+	(3367, 374400),
+	(2030, 374400),
+	(2046, 374400),
+	(3366, 374400),
+	(2037, 374400),
+	(2043, 374400),
+	(2013, 374400),
+	(3370, 389520),
+	(2086, 389520),
+	(2051, 389520),
+	(2008, 389520),
+	(3359, 389520),
+	(3377, 389520),
+	(3373, 389520),
+	(2092, 404640),
+	(3378, 404640),
+	(2104, 404640),
+	(3389, 404640),
+	(2090, 404640),
+	(3397, 404640),
+	(2091, 404640),
+	(3388, 404640),
+];
```

Comment:
```suggestion
const LEASES: [(u32, u32); 46] = [
	(2094, 298800),
	(2040, 298800),
	(2035, 359280),
	(3344, 344160),
	(3370, 389520),
	(3367, 374400),
	(2086, 389520),
	(2032, 359280),
	(3333, 298800),
	(2051, 389520),
	(2106, 298800),
	(3369, 313920),
	(2008, 389520),
	(2025, 359280),
	(2000, 313920),
	(2092, 404640),
	(2002, 359280),
	(3359, 389520),
	(2030, 374400),
	(3378, 404640),
	(2104, 404640),
	(2046, 374400),
	(3345, 344160),
	(3340, 344160),
	(3338, 329040),
	(2004, 329040),
	(3377, 389520),
	(3373, 389520),
	(2031, 344160),
	(3389, 404640),
	(3366, 374400),
	(2037, 374400),
	(2034, 359280),
	(2090, 404640),
	(3346, 344160),
	(2012, 359280),
	(3397, 404640),
	(2043, 374400),
	(2091, 404640),
	(2093, 298800),
	(2026, 344160),
	(3388, 404640),
	(2101, 298800),
	(3354, 359280),
	(2006, 344160),
	(2013, 374400),
];
```
Same values but ordered by the core id from the workplan. I've obtained the list with this script:
```js
const { ApiPromise, WsProvider } = require('@polkadot/api');

function parse_pjs_int(input) {
    return parseInt(input.replace(/,/g, ''));
}

const timeslices = {
    2094: 298800,
	2040: 298800,
	3333: 298800,
	2106: 298800,
	2093: 298800,
	2101: 298800,
	3369: 313920,
	2000: 313920,
	3338: 329040,
	2004: 329040,
	3344: 344160,
	3345: 344160,
	3340: 344160,
	2031: 344160,
	3346: 344160,
	2026: 344160,
	2006: 344160,
	2035: 359280,
	2032: 359280,
	2025: 359280,
	2002: 359280,
	2034: 359280,
	2012: 359280,
	3354: 359280,
	3367: 374400,
	2030: 374400,
	2046: 374400,
	3366: 374400,
	2037: 374400,
	2043: 374400,
	2013: 374400,
	3370: 389520,
	2086: 389520,
	2051: 389520,
	2008: 389520,
	3359: 389520,
	3377: 389520,
	3373: 389520,
	2092: 404640,
	3378: 404640,
	2104: 404640,
	3389: 404640,
	2090: 404640,
	3397: 404640,
	2091: 404640,
	3388: 404640,
};

async function main() {
    const coretime_chain_api = await ApiPromise.create({ provider: new WsProvider('wss://polkadot-coretime-rpc.polkadot.io') });

    const workplan = await coretime_chain_api.query.broker.workplan.entries();

    const data = workplan.map(([{ args: [key] }, value]) => {
        // console.log('key/value', key.toHuman(), value.toHuman());
        const core_id = parse_pjs_int(key.toHuman()[1]);
        const para_id = parse_pjs_int(value.toHuman()[0].assignment.Task);

        // console.log("core_id: %d", core_id);
        // console.log("para_id: %s", para_id);

        return [core_id, para_id];
    });
    const sorted_data = data.sort((a, b) => a[0] - b[0]);
    sorted_data.forEach((item) => {
        console.log("(%d, %d),", item[1], timeslices[item[1]]);
    });
    // console.log(sorted_data);
}

main().catch(console.error).finally(() => process.exit());
```